mldsa/polyvec.h

```diff
@@ -72,11 +72,18 @@ __contract__(
  * Name:        polyvecl_ntt
  *
  * Description: Forward NTT of all polynomials in vector of length MLDSA_L.
- *Output coefficients can be up to 16*MLDSA_Q larger than input coefficients.
+ *              Coefficients can grow by 8*MLDSA_Q in absolute value.
  *
  * Arguments:   - polyvecl *v: pointer to input/output vector
  **************************************************/
-void polyvecl_ntt(polyvecl *v);
+void polyvecl_ntt(polyvecl *v)
+__contract__(
+  requires(memory_no_alias(v, sizeof(polyvecl)))
+  requires(forall(k0, 0, MLDSA_L, array_abs_bound(v->vec[k0].coeffs, 0, MLDSA_N, MLDSA_Q)))
+  assigns(memory_slice(v, sizeof(polyvecl)))
+  ensures(forall(k1, 0, MLDSA_L, array_abs_bound(v->vec[k1].coeffs, 0, MLDSA_N, MLD_NTT_BOUND)))
+);
+
```

Comment:
Looks good to me! I had got to a similar state with the proof I started on this

```
__contract__(
  requires(memory_no_alias(v, sizeof(polyvecl)))
  requires(forall(i, 0, MLDSA_L,
    array_abs_bound(v->vec[i].coeffs, 0, MLDSA_N, MLDSA_Q)))
  assigns(object_whole(v))
  ensures(forall(i, 0, MLDSA_L,
    array_abs_bound(v->vec[i].coeffs, 0, MLDSA_N, MLD_NTT_BOUND)))
);
```
But I had abandoned due to issue 8617